### ISSUE TITLE
[META] [UX] admin menu entries order is a mess.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#523]
The "home" menu is fine (all entries listed alphabetically):
- Home
  - Flush all caches
    - Administration bar
    - CSS and JavaScript
    - Menu
    - Page and else
    - Theme registry
    - Update data
  - Run cron
  - Run updates
- Content
  - Add content
    - Page
    - Post
    - ...[custom content types]...
  - Find content
  - Comments
    - Published comments
    - Unapproved comments ([count])
- People
  - Add user account
  - Find user accounts
- Functionality
  - Install module ~~#399~~
  - Modules
  - Update
  - Uninstall
- Appearance
  - Install theme ~~#399~~
  - Themes
  - Bartik settings
  - ...[more themes' settings]...
  - Update
- Structure
  - Content types
    - Add content type
    - List content types #525
    - Article
      - Delete
      - Edit
      - Manage fields
        - Image
          - Field settings
          - Widget type
          - Edit
        - Tags
          - Field settings
          - Widget type
          - Edit
        - Body
          - Field settings
          - Widget type
          - Edit
          - Manage display
            - Default
            - Teaser
    - Basic page
      - Delete
      - Edit
      - Manage fields
        - Body
          - Field settings
          - Widget type
          - Edit
      - Manage display
        - Default
        - Teaser
    - ...[more custom content types]...
  - Custom blocks
    - Add block
    - List blocks #525
  - Layouts
    - Add layout ~~#399~~
    - List layouts #525
  - Menus
    - Add menu ~~#399~~
    - List menus #525
    - Account menu
      - Add link ~~#399~~
      - List links #525
      - Delete menu
      - Edit menu
    - Administration menu
      - Add link ~~#399~~
      - List links #525
      - Delete menu
      - Edit menu
    - Primary navigation
      - Add link ~~#399~~
      - List links #525
      - Delete menu
      - Edit menu
    - Settings
  - Taxonomy
    - Add vocabulary ~~#399~~
    - List vocabularies #525
  - Views #521
    - Add view ~~#399~~
    - Import
    - List views #525
    - Admin Content (Content)
      - Edit view
    - Settings
      - Basic
      - Advanced
    - Taxonomy term (Content)
      - Edit view
    - Recent comments (Comment)
      - Edit view
    - Administer users (User)
      - Edit view
    - Front page (Content)
      - Edit view
- Configuration
  - Administration
    - Administration bar
  - Content authoring
    - Text formats
      - Add text format ~~#399~~
      - Filtered HTML
        - Disable text format
      - Full HTML
        - Disable text format
      - List text formats #525
      - Plain text
  - Development
    - Performance
    - Logging and errors
    - Maintenance mode
    - Configuration management
      - Synchronize
      - Configuration diff
      - Single Import/Export
        - Import
        - Export
      - Full Import/Export
        - Import
        - Export
  - Media
    - File system
    - Image styles
      - Add image style ~~#399~~ #948
      - List image styles #525
    - Image toolkit
  - People
    - Permissions
    - Roles
    - Manage fields
      - Manage fields
      - Manage display
        - Default
    - Account settings
  - Regional and language
    - Regional settings
    - Date and time formats
      - Add date format ~~#399~~
  - Search and metadata
    - Search settings
    - URL aliases
      - Add URL alias ~~#399~~ #948
      - List URL aliases #525
    - Clean URLs
  - System
    - Site information
    - Cron
  - Web services
    - RSS publishing
- Reports
  - Available updates
    - Install new module or theme
      - List available updates #525
      - Update
      - Settings
    - Status report
    - Recent log messages
    - Fields #525
      - List fields #525
      - Used in views
    - Top 'access denied' errors
    - Top 'page not found' errors
    - Top search phrases
    - Views plugins

**Comment by klonos** [comment]
@quicksketch said in #521:

> We're currently special casing the following:

- Text formats
- Menus
- Node types
- Listing fields within entity types (node types, vocabularies)
- Vocabularies
- Views

By "special casing" I mean we're augmenting the "normal" menu (as provided by hook_menu() and the Administration menu configuration) with dynamic entries. All of these implementations are in admin_bar.map.inc.

**Comment by klonos** [comment]
...actually, we do not list the vocabularies and any of their available their actions/operations under ```Structure``` -> ```Taxonomy``` at all.

**Comment by klonos** [comment]
Before I open a separate issue for them, I'd like to bring the attention to the modules and themes menus/pages...

So we have:

- Functionality (```/admin/modules```)
  - Install new module (```/admin/modules/install```)
  - Modules (```/admin/modules/list```)
  - Update (```/admin/modules/update```)
  - Uninstall (```/admin/modules/uninstall```)

and:

- Appearance (```/admin/appearance```)
  - Install new theme (```/admin/appearance/install```)
  - Themes (```/admin/appearance/list```)
  - Bartik (```/admin/appearance/settings/bartik```)
  - Update (```/admin/appearance/update```)

Now I've followed all the issues/discussions over at drupal.org about changing the "modules" and "themes" user-facing text to "functionality" and "appearance" respectively for D8 and I believe it was a good (or at least justifiable) decision. But why change the path (that is not a user-facing thing) for themes while keeping the path for modules? Is this something we should revert in backdrop so we are more inline with D7? ...again, I'm talking about the ```appearance``` path part - not the "Appearance" user-facing text.

Also, I think that "Bartik" (as well as all other themes) entries should go under ```Appearance``` -> ```Themes```.

**Comment by klonos** [comment]
...besides bad the bad DX of having modules have a "Functionality" user text while themes having both the text as well as the path be "appearance", consider the confusion if in the future we decide to change the Appearance UI text to something else.

**Comment by klonos** [comment]
> I propose adding a hierarchical list of *all* Admin Bar links to the top of this issue...

Done.

> As we decide on what should be changed, we update the list to reflect those changes. Once the list is setup as we'd like it to be, we can start a PR for the change.

I think that this will get out of hand as this list is huge. Lets just leave it like it is for reference and simply split off to separate issues so we can move in small steps. One additional reason to move this way is that it is not only the menu structure that is wrong but also the underlying pages/tabs/lists.

> We can put a link next to each sub-menu in the above list for the GitHub issue where that menu is being discussed...

Already doing so as I file issues.

**Comment by BWPanda** [comment]
My take on the overall ordering of Admin Bar menu items:

1) Where a sub-menu item links to the same page as its parent menu item, that sub-menu item should be listed first in the sub-menu.
e.g. 'People' links to the page that lists all user accounts. 'People > Find user accounts' links to the same page. Therefore 'People > Find user accounts' should be listed first in the 'People' menu.

2) To standardise, as mentioned above, there should be no links to things in the menu that users can create a potentially unlimited amount of; just links for listing these things, adding new ones, etc.
e.g. Links to specific content types, menus, text formats, views, etc.

3) The sub-menu item ordering of menus from my 2nd point above should be:
  1. List/find (as per my 1st point)
  2. Add/install/create
  3. Other (e.g. settings, import, uninstall, etc.)

4) All other menu items should be listed alphabetically.
e.g. 'Configuration > Media' sub-menu.

**Comment by klonos** [comment]
> 1) Where a sub-menu item links to the same page as its parent menu item, that sub-menu item should be listed first in the sub-menu.

In #525 I just threw the idea of dropping the "List xyz" items altogether if their respective parent item links to the very same page.

> 2) To standardise, as mentioned by others above, there should be no links to things in the menu that users can create a potentially unlimited amount of;...

I don't like this idea particularly. I do see the point if the amount of items listed is really big, but how many sites out there have dozens of content types or views? Are we perhaps accounting for the 20% this way?

**Comment by klonos** [comment]
...just updated the menu in the issue summary with the outcome of #399 and how it should be after #525

**Comment by klonos** [comment]
...and #948

**Comment by jenlampton** [comment]
just filed PRs for all the remaining items in this issue.  Here's to hoping we can get them all cleaned up for 1.3.0 :)

**Comment by jenlampton** [comment]
It looks like all the sub-issues here are already closed. Going to close this parent issue too. I think the one we should be focusing on is https://github.com/backdrop/backdrop-issues/issues/2260 and https://github.com/backdrop/backdrop-issues/issues/2156
